src/granule_entry/handler.py

```diff
@@ -0,0 +1,261 @@
+        logger.error("AUX_DATA_BUCKET_NAME environment variable not set")
+        return False
+
+    year = granule_id.begin_datetime.strftime("%Y")
+    ydoy = granule_id.begin_datetime.strftime("%Y%j")
+
+    vj_pattern = f"lasrc_aux/LADS/{year}/VJ104ANC.A{ydoy}"
+    vj_exists = key_pattern_exists(bucket, vj_pattern)
+
+    vnp_pattern = f"lasrc_aux/LADS/{year}/VNP04ANC.A{ydoy}"
+    vnp_exists = key_pattern_exists(bucket, vnp_pattern)
+
+    return bool(vj_exists or vnp_exists)
+
+
+@tracer.capture_method
+@tracer.capture_method
+def process_record(sqs_body: str) -> None:
+    """Process a single SQS message body containing S3 event notifications.
+
+    Args:
+        sqs_body: SQS message body containing SNS-wrapped S3 event notifications
+    """
+    logs_bucket = os.environ["PROCESSING_BUCKET_NAME"]
+    logs_prefix = os.environ.get("PROCESSING_BUCKET_LOG_PREFIX", "logs")
+
+    granule_logger = GranuleLoggerService(
+        bucket=logs_bucket,
+        logs_prefix=logs_prefix,
+    )
+
+    s3_records = parse_s3_sns_message(sqs_body)
+
+    for s3_record in s3_records:
+        s3_info = s3_record["s3"]
+        bucket_name = s3_info["bucket"]["name"]
+        object_key = s3_info["object"]["key"]
+
+        safe_id = extract_safe_id_from_s3_key(object_key)
+        granule_id_str = convert_safe_id_to_hls_id(safe_id)
+        granule_id = GranuleId.from_str(granule_id_str)
+
+        granule_event = GranuleProcessingEvent(
+            granule_id=granule_id_str,
+            source_granule_id=safe_id,
+            attempt=0,
+        )
+
+        aux_data = check_aux_data(granule_id)
+        if aux_data:
+            submit_job(
+                granule_id=granule_id_str,
+                source_granule_id=safe_id,
+                granule_logger=granule_logger,
+            )
+        else:
+            logger.info(
+                "Marking granule as AWAITING",
+                extra={
+                    "granule_id": granule_event.granule_id,
+                    "s3_bucket": bucket_name,
+                    "s3_key": object_key,
+                },
+            )
+            granule_logger.put_event(
+                event=granule_event,
+                state=ProcessingState.AWAITING,
+            )
+
+
+@logger.inject_lambda_context
+@tracer.capture_lambda_handler
+@metrics.log_metrics
+def handler(event: dict[str, Any], context: LambdaContext) -> None:
+    """Lambda handler for processing a single S3 event notification.
+
+    Args:
+        event: SQS event containing a single SNS-wrapped S3 notification
+        context: Lambda context object
+    """
+    # Extract the single SQS record from the event
+    records = event.get("Records", [])
+    if not records:
+        logger.warning("No records found in event")
+        return
+
+    if len(records) > 1:
+        logger.warning(
+            "Multiple records received, only processing the first",
+            extra={"record_count": len(records)},
+        )
```

Comment:
I saw `batch_size=100` over in the CDK code connecting the event notifications to this Lambda so I'd expect the records to be `<= 100` in length.

Is the limit of `1` just for debugging right now, or is that more intentional? Is the AWS Lambda powertools doing something here maybe?